src/dotnet-roslyn-tools/CreateReleaseTags/CreateReleaseTags.cs

```diff
@@ -98,30 +99,55 @@ public static async Task<int> CreateReleaseTagsAsync(RoslynToolsSettings setting
         }
         catch (Exception ex)
         {
-            logger.LogError(ex, "Unable to open Roslyn repo. Please run from you Roslyn repo directory.");
+            logger.LogError(ex, "Unable to open repo. Please run from your repo directory for the product in question.");
         }
 
         return 1;
     }
 
-    private static async Task<RoslynBuildInformation?> TryGetRoslynBuildForReleaseAsync(VisualStudioVersion release, AzDOConnection vsConnection, AzDOConnection connection)
+    private static async Task<RoslynBuildInformation?> TryGetBuildInfoForReleaseAsync(IProduct product, VisualStudioVersion release, AzDOConnection vsConnection, AzDOConnection connection)
     {
         try
         {
-            var (branchName, buildNumber) = await TryGetRoslynBranchAndBuildNumberForReleaseAsync(release, vsConnection);
+            var url = await VisualStudioRepository.GetUrlFromComponentJsonFileAsync(release.CommitSha, GitVersionType.Commit, vsConnection, product.ComponentJsonFileName, product.ComponentName);
+            if (url is null)
+            {
+                return default;
+            }
+
+            // First we try to get the info we want from the build directly, if we can find it
+            var buildNumber = VisualStudioRepository.GetBuildNumberFromUrl(url);
+            var pipelineName = product.GetBuildPipelineName(connection.BuildProjectName);
+            if (pipelineName is not null)
+            {
+                var builds = await connection.TryGetBuildsAsync(pipelineName, buildNumber);
+                if (builds is not null)
+                {
+                    foreach (var build in builds)
```

Comment:
When can there be more than one here? And the first one w/ a non-empty source branch is just always correct?

Reply:
I don't understand how or why, but yes, in the past there have been multiple builds on the same pipeline with the same build number.

Reply:
That seems terrifying. Should we just throw in that case and if/when that ever happens again we can then figure out what the right action is? My concern is that might tag the wrong commit in that case and we won't actually know if this did the right thing.

Reply:
Dug up some context on this (surprisingly, Teams search actually worked!): https://teams.microsoft.com/l/message/19:ed7a508bf00c4b088a7760359f0d0308@thread.skype/1621985340130?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=4ba7372f-2799-4677-89f0-7a1aaea3706c&parentMessageId=1621985340130&teamName=.NET%20Developer%20Experience&channelName=InfraSwat&createdTime=1621985340130

So this was a bug, and was probably a one-off issue that it happened, but given that a) obviously AzDo doesn't care if you tell it something has the same build number as something else and b) this tool looks through historical builds so the bugged build could still be in play, I would be hesitant to remove this. When it did come up (it was in running the `vsbranchinfo` tool to do a VS Mac insertion) the commit that it found was correct, but I don't remember much else about it.

Reply:
Even if we take for granted that we need to support this case, there's still no argument here for why this is picking the right one versus the wrong one. I'd also imagine that doesn't matter anymore since we wouldn't be trying to create release tags for either of those builds unless we actually shipped one. And we could easily resolve the problem by just deleting one of the historical builds that we know wasn't shipped.